src/background/browser-detection.ts

```diff
@@ -0,0 +1,20 @@
+function testBrowser(browser: SupportedBrowser, navigator: typeof window.navigator): null | BrowserInfo {
```

Comment:
I like the idea in theory but in practice that means the user has to wait for the iframe to fail, which involves a network round trip, and there's more work on our end to figure out that that's what happened. Sniffing the user agent tells us a lot about what to expect from the browser behavior. Since we're only supporting 2 browsers for now (we don't have to do anything with mobile browsers) I think it's ok for now